tests/rptest/tests/license_upgrade_test.py

```diff
@@ -0,0 +1,98 @@
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+import os
+import re
+
+from rptest.services.admin import Admin
+from ducktape.utils.util import wait_until
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import SISettings
+from rptest.services.cluster import cluster
+from requests.exceptions import HTTPError
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+from rptest.services.redpanda_installer import RedpandaInstaller, wait_for_num_versions
+
+
+def license_env_ci():
+    license = os.environ.get("REDPANDA_SAMPLE_LICENSE", None)
+    if license is None:
+        is_ci = os.environ.get("CI", "false")
+        assert is_ci == "false"
+        self.logger.info(
+            "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+        return None
+    return license
+
+
+class UpgradeToLicenseChecks(RedpandaTest):
+    """
+    """
+    LICENSE_CHECK_INTERVAL_SEC = 1
+
+    def __init__(self, test_context):
+        super(UpgradeToLicenseChecks, self).__init__(test_context=test_context,
+                                                     num_brokers=3,
+                                                     enable_installer=True,
+                                                     si_settings=SISettings())
+        self.installer = self.redpanda._installer
+        self.installed_version = (22, 1, 4)
+        self.admin = Admin(self.redpanda)
+
+    def setUp(self):
+        self.installer.install(self.redpanda.nodes, self.installed_version)
+        super(UpgradeToLicenseChecks, self).setUp()
+
+    @cluster(num_nodes=3, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_basic_upgrade(self):
+        # Modified environment variables apply to processes restarted from this point onwards
+        self.redpanda.set_environment({
+            '__REDPANDA_LICENSE_CHECK_INTERVAL_SEC':
+            f'{UpgradeToLicenseChecks.LICENSE_CHECK_INTERVAL_SEC}'
+        })
+
+        license = license_env_ci()
+        if license is None:
+            # skip the test
+            return
+
+        unique_versions = wait_for_num_versions(self.redpanda, 1)
+        assert 'v22.1.4' in unique_versions, unique_versions
+
+        # These logs can't exist in v22.1.4 but double check anyway...
+        assert self.redpanda.search_log("Enterprise feature(s).*") is False
+
+        # Update one node to newest version
+        self.installer.install([self.redpanda.nodes[0]],
+                               RedpandaInstaller.HEAD)
+        self.redpanda.restart_nodes([self.redpanda.nodes[0]])
+        unique_versions = wait_for_num_versions(self.redpanda, 2)
+
+        try:
+            # Ensure a valid license cannot be uploaded in this cluster state
+            assert self.admin.put_license(license)
+        except HTTPError as e:
+            assert e.response.status_code == 400
+
```

Comment:
Here could we wait past the license interval and check that we still don't see any logs in the mixed-version cluster (maybe belongs after the other patches).

Reply:
Yeah the search log method searches the entire log though. I'd need to modify it to look for logs written since now.

Reply:
If the first check passes and we don't have any bad logs now, I dont think we'd have to do any modifying since any new bad log would necessarily be after now.

Reply:
ah i misread your comment, yes good point +1